src/main/java/io/github/openDFDL/TraceListener.java

```diff
@@ -0,0 +1,54 @@
+/**********************************************************************
+* Licensed Materials - Property of IBM
+*
+* (C) Copyright IBM Corp. 2012
+*
+* US Government Users Restricted Rights - Use, duplication, or
+* disclosure restricted by GSA ADP Schedule Contract with IBM Corp.
+**********************************************************************/
+package io.github.openDFDL;
+
+/**
+ * Example implementation of the DFDL IDFDLTrace interface.
+ * This simple implementation prints to stdout on each call.
+ */
+
+import com.ibm.dfdl.processor.trace.IDFDLTrace;
+import com.ibm.dfdl.processor.trace.IDFDLUserTraceListener;
+import com.ibm.dfdl.processor.trace.IDFDLServiceTraceListener;
+
+public class TraceListener implements IDFDLUserTraceListener, IDFDLServiceTraceListener {
+
+	public void info(IDFDLTrace traceItem) {
+		System.err.println("TraceListener : info() "+traceItem.getMessage() );
+	}
+
+	public void warning(IDFDLTrace traceItem) {
+		System.err.println("TraceListener : warning() "+traceItem.getMessage() );
+	}
+	
+	public void error(IDFDLTrace traceItem) {
+		System.err.println("TraceListener : error() "+traceItem.getMessage() );
+	}
+	
+	public void fatal(IDFDLTrace traceItem) {
+		System.err.println("TraceListener : fatal() "+traceItem.getMessage() );
+	}
+
+	@Override
+	public void detail(IDFDLTrace traceItem) {
+		System.err.println("TraceListener : detail() "+traceItem.getMessage() );
+	}
+
+	@Override
+	public void entry(IDFDLTrace traceItem) {
+		// System.err.println("TraceListener : entry() "+traceItem.getMessage() );
```

Comment:
Any reason why this and "exit" below have commented out print statements, but the rest don't?

Reply:
Yes. They aren't desirable to see in a trace. They're really in case you implement indenting. For a basic simple trace output you don't need to see these events.